File: android/app/src/main/java/br/edu/ifba/mybeerapp/model/Modelo.java
package br.edu.ifba.mybeerapp.model;

import br.edu.ifba.mybeerapp.model.annotations.DBField;
import br.edu.ifba.mybeerapp.model.interfaces.IModel;

public class Modelo implements IModel
{
    @DBField
    private int id;
    @DBField
    private String nome;
    @DBField
    private Integer volume; // ml

    public Modelo(){}

    public Modelo(String nome, int volume) {
        this.nome = nome;
        this.volume = volume;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Override
    public IModel clone() {
        return null;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Integer getVolume() {
        return volume;
    }

    public void setVolume(Integer volume) {
        this.volume = volume;
    }

    public String toString()
    {
        return String.valueOf(this.id);
    }
}

File: flutter_mybeerapp/lib/models/modelo.dart
class Modelo {
  final int id;
  final String nome;
  final int volume;

  Modelo({this.id, this.nome, this.volume});

  factory Modelo.fromJson(Map<String, dynamic> json) {
    return Modelo(
        id: json['id'] as int,
        nome: json['nome'],
        volume: json['volume'] as int
    );
  }

  Map<String, dynamic> toJSON() {
    return {
      "id": this.id,
      "nome": this.nome,
      "volume": this.volume
    };
  }

  @override
  String toString() {
    return this.nome + " - " + this.volume.toString() + "ml";
  }

  bool operator == (o) => o is Modelo && o.id == id;

}